src/panoramic/cli/connections.py

```diff
@@ -0,0 +1,281 @@
+from typing import Any, Dict, Optional, Tuple
+
+import click
+import hologram
+import yaml
+
+from panoramic.cli.config.storage import read_config, update_config
+from panoramic.cli.errors import ConnectionNotFound
+from panoramic.cli.paths import Paths
+from panoramic.cli.print import echo_info
+
+CONNECTION_KEYS = [
+    'type',
+    'user',
+    'host',
+    'port',
+    'password',
+    'database',
+    'schema',
+    'warehouse',
+    'account',
+    'project',
+    'key_file',
+]
+
+
+def create_connection_command(
+    name,
+    type,
+    user,
+    host,
+    port,
+    password,
+    password_stdin,
+    database,
+    schema,
+    warehouse,
+    account,
+    project,
+    key_file,
+    no_test,
+):
+    """CLI command. Create new connection."""
+    connections = Connections.load()
+    if name in connections:
+        raise click.ClickException(f'Connection with name "{name}" already exists.')
+
+    if password_stdin:
+        password = click.prompt('Enter password: ', hide_input=True, type=str)
+
+    new_connection = {
+        'type': type,
+        'user': user,
+        'host': host,
+        'port': port,
+        'password': password,
+        'database': database,
+        'schema': schema,
+        'warehouse': warehouse,
+        'account': account,
+        'project': project,
+        'key_file': key_file,
+    }
+
+    credentials, error = get_dialect_credentials(new_connection)
+    if error != '':
+        raise click.ClickException(f'Failed to create connection: {error}')
+
+    if not no_test:
+        ok, error = Connections.test(credentials)
+        if not ok:
+            raise click.ClickException(f'Failed to create connection: {error}')
+
+    # Empty values are important for DBT credentials verification, but we don't need to store empty values in config.
+    connections[name] = {}
+    for key, value in new_connection.items():
+        if value:
+            connections[name][key] = value
+
+    Connections.save(connections)
+    echo_info('Connection was successfully created!')
+
+
+def update_connection_command(
+    name,
+    type,
+    user,
+    host,
+    port,
+    password,
+    password_stdin,
+    database,
+    schema,
+    warehouse,
+    account,
+    project,
+    key_file,
+    no_test,
+):
+    """CLI command. Update specific connection."""
+    connections = Connections.load()
+    if name not in connections:
+        raise click.ClickException(f'Connection with name "{name}" not found.')
+
+    if password_stdin:
+        password = click.prompt('Enter password: ', hide_input=True, type=str)
+
+    new_connection = connections[name].copy()
+    if type:
```

Comment:
Can we check against None here and below to avoid accidentally not setting a value when the value is falsy?

Reply:
Agree with what @mark mentioned 
```suggestion
    if type is not None:
```

Reply:
Fixed.